apps/pie-microsite/src/_11ty/shortcodes/tokensTable.js

```diff
@@ -130,6 +149,138 @@ const createTokensList = listElements => `<div class="c-tokensTable-row u-spacin
   ${listElements.join('')}
 </ul>`;
 
+/**
+ * Creates a tokens list for a given category
+ * @param {*} tokens 
+ * @param {string} path 
+ * @param {string} category 
+ * @param {string} tokenType 
+ * @returns 
+ */
+const buildTokensListForCategory = (tokens, path, category, tokenType) => {
+    const tokenTypeMetadata = getTokenTypeMetadata(path);
+    const tokensForCategory = getTokensForCategory(category, tokenTypeMetadata);
+
+    // create a list item for the current token
+    const tokenListElements = tokensForCategory.map(key => buildTokenListElements({
+        token: tokens[key],
+        tokenScssName: createScssTokenName(key, tokenType),
+        tokenDisplayName: createTokenDisplayName(key, tokenType),
+        tokenType,
+        tokenMetadata: tokenTypeMetadata[key]
+    }));
+
+    return buildTokensList(tokenListElements);
+};
+
+/**
+ * Builds a categorised list of tokens
+ * @returns {string}
+ */
+const buildCategorisedLists = ({
+    path, tokenType, tokens 
+}) => {
+    const categories = objectHelpers.getObjectPropertyByPath(pieTokenCategories, path);
+
+    // for each category, create an h2 and a list of token elements to render
+    const categoryTokenLists = Object.keys(categories).map(category => {
+        const heading = `<h2>${categories[category].displayName}</h2>`;
+        const tokensList = buildTokensListForCategory(tokens, path, category, tokenType);
+
+        // returns a 'chunk' of the tokens table page (a heading, the column headers, list of tokens and an option HR element)
+        return `${heading}${tokensList}`;
+    });
+
+    // all 'chunks' of the tokens table page HTML in a single string
+    return categoryTokenLists.join('<hr />');
+};
+
+/**
+ * Gets all the metadata associated with tokens of a given type such as colour.
+ * @param {string} path 
+ * @returns {object}
+ */
+const getTokenTypeMetadata = path => objectHelpers.getObjectPropertyByPath(pieTokensMetadata, path);
+
+/**
+ * Gets all tokens for a given category such as 'orange'
+ * @param {string} category 
+ * @param {object} tokenTypeMetadata 
+ * @returns {string[]}
+ */
+const getTokensForCategory = (category, tokenTypeMetadata) => Object
+    .keys(tokenTypeMetadata)
+    .filter(token => tokenTypeMetadata[token].category === category);
+
+
+/**
+ * Gets all subcategory keys for a given parent category
+ * @param {object} tokenTypeCategories 
+ * @param {string} parentCategoryKey 
+ * @returns 
+ */
+const getSubcategoriesForParentCategory = (tokenTypeCategories, parentCategoryKey) => Object
+          .keys(tokenTypeCategories)
+          .filter(categoryKey => tokenTypeCategories[categoryKey].parentCategory === parentCategoryKey);
+
+/**
+ * Builds a list of tokens that are categorised by parent and subcategory
+ * @param {object} config 
+ * @returns {string}
```

Comment:
An example of the return would be useful – looks like it's a list of tokens separate by `<hr />` tags, but an example here would be useful.